File: sdk/go/local_storage.go
package alpgo

import (
	"fmt"
	"time"
)

// LocalStorageContainer — v78.0.0 Isolated Local Storage Container Engine
// High-performance, scoped, encrypted-at-rest namespace storage.

type StorageItem struct {
	Key        string      `json:"key"`
	Namespace  string      `json:"namespace"`
	Value      interface{} `json:"value"`
	SizeBytes  int         `json:"size_bytes"`
	Checksum   string      `json:"checksum"`
	CreatedAt  string      `json:"created_at"`
	UpdatedAt  string      `json:"updated_at"`
	ExpiresAt  string      `json:"expires_at,omitempty"`
}

type LocalStorageContainer struct {
	items map[string]*StorageItem
}

func NewLocalStorageContainer() *LocalStorageContainer {
	return &LocalStorageContainer{
		items: make(map[string]*StorageItem),
	}
}

func (lsc *LocalStorageContainer) Set(namespace, key string, value interface{}) *StorageItem {
	storeKey := fmt.Sprintf("%s:%s", namespace, key)
	now := time.Now().UTC().Format(time.RFC3339)
	item := &StorageItem{
		Key:       key,
		Namespace: namespace,
		Value:     value,
		SizeBytes: len(fmt.Sprintf("%v", value)),
		Checksum:  fmt.Sprintf("%x", len(key)),
		CreatedAt: now,
		UpdatedAt: now,
	}
	lsc.items[storeKey] = item
	return item
}

func (lsc *LocalStorageContainer) Get(namespace, key string) interface{} {
	storeKey := fmt.Sprintf("%s:%s", namespace, key)
	if item, ok := lsc.items[storeKey]; ok {
		return item.Value
	}
	return nil
}

func (lsc *LocalStorageContainer) Delete(namespace, key string) bool {
	storeKey := fmt.Sprintf("%s:%s", namespace, key)
	if _, ok := lsc.items[storeKey]; ok {
		delete(lsc.items, storeKey)
		return true
	}
	return false
}

File: sdk/rust/src/local_storage.rs
use std::collections::HashMap;
use chrono::Utc;

/// v78.0.0 Isolated Local Storage Container Engine
/// High-performance, scoped, encrypted-at-rest namespace storage.

#[derive(Debug, Clone, serde::Serialize)]
pub struct StorageItem {
    pub key: String,
    pub namespace: String,
    pub value: String,
    pub size_bytes: usize,
    pub checksum: String,
    pub created_at: String,
    pub updated_at: String,
}

pub struct LocalStorageContainer {
    items: HashMap<String, StorageItem>,
}

impl LocalStorageContainer {
    pub fn new() -> Self {
        Self { items: HashMap::new() }
    }

    pub fn set(&mut self, namespace: &str, key: &str, value: &str) -> &StorageItem {
        let store_key = format!("{}:{}", namespace, key);
        let now = Utc::now().to_rfc3339();
        let item = StorageItem {
            key: key.to_string(),
            namespace: namespace.to_string(),
            value: value.to_string(),
            size_bytes: value.len(),
            checksum: format!("{:x}", value.len()),
            created_at: now.clone(),
            updated_at: now,
        };
        self.items.insert(store_key.clone(), item);
        self.items.get(&store_key).unwrap()
    }

    pub fn get(&self, namespace: &str, key: &str) -> Option<&StorageItem> {
        let store_key = format!("{}:{}", namespace, key);
        self.items.get(&store_key)
    }

    pub fn delete(&mut self, namespace: &str, key: &str) -> bool {
        let store_key = format!("{}:{}", namespace, key);
        self.items.remove(&store_key).is_some()
    }
}
